cmd/account/mgmt/account-assign.go

```diff
@@ -183,6 +195,117 @@ func (o *accountAssignOptions) tagAccount(accountIdInput string) error {
 	return nil
 }
 
+func (o *accountAssignOptions) buildAccount(seedVal int64) (string, error) {
+
+	fmt.Println("Creating account")
+
+	orgOutput, orgErr := o.createAccount(seedVal)
+	if orgErr != nil {
+		// If email already exists retry up to 100 times
+		if orgErr == ErrEmailAlreadyExist {
+			err := retry(100, 2*time.Second, func() (err error) {
+				orgOutput, err = o.createAccount(seedVal)
+				return
+			})
+			if err != nil {
+				return "", err
+			}
+		}
+		return "", orgErr
+	}
+
+	fmt.Println("account created successfully")
+
+	return *orgOutput.CreateAccountStatus.AccountId, nil
+}
+
+func retry(attempts int, sleep time.Duration, callback func() error) (err error) {
+	for i := 0; ; i++ {
+		err = callback()
+		if err == nil {
+			return
+		}
+
+		if i >= (attempts - 1) {
+			break
+		}
+		time.Sleep(sleep)
+
+	}
+	return fmt.Errorf("after %d attempts, last error: %s", attempts, err)
+}
+
+var ErrAwsAccountLimitExceeded error = fmt.Errorf("ErrAwsAccountLimitExceeded")
+var ErrEmailAlreadyExist error = fmt.Errorf("ErrEmailAlreadyExist")
+var ErrAwsInternalFailure error = fmt.Errorf("ErrAwsInternalFailure")
+var ErrAwsTooManyRequests error = fmt.Errorf("ErrAwsTooManyRequests")
+var ErrAwsFailedCreateAccount error = fmt.Errorf("ErrAwsFailedCreateAccount")
+
+func (o *accountAssignOptions) createAccount(seedVal int64) (*organizations.DescribeCreateAccountStatusOutput, error) {
+
+	rand.Seed(seedVal)
+	randStr := RandomString(6)
+	accountName := "osd-creds-mgmt" + "+" + randStr
+	email := accountName + "@redhat.com"
+
+	createInput := &organizations.CreateAccountInput{
+		AccountName: aws.String(accountName),
+		Email:       aws.String(email),
+	}
+
+	createOutput, err := o.awsClient.CreateAccount(createInput)
+	if err != nil {
+		return &organizations.DescribeCreateAccountStatusOutput{}, err
+	}
+
+	describeStatusInput := &organizations.DescribeCreateAccountStatusInput{
+		CreateAccountRequestId: createOutput.CreateAccountStatus.Id,
```

Comment:
This seems to be the problem: there is no value for `CreateAccountRequestId`

unreadable: could not read string len protocol error E08 during memory read for packet $mc4834870246c8b48,1...